File: problems/LeetCode/blind150/two_pointers/p167.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        p, q = 0, len(numbers) - 1

        while p < q:
            if numbers[p] + numbers[q] == target:
                return [p+1, q+1]

            if numbers[p] + numbers[q] < target:
                p += 1
            else:
                q -= 1

        return None


if __name__ == '__main__':
    s = Solution()
    print(s.twoSum([2, 3, 4, 15], 6))

File: LeetCode-Java/p167.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int p = 0;
        int n = numbers.length - 1;

        while (p < n) {
            int sum = numbers[p] + numbers[n];
            
            if (sum == target) {
                return new int[]{p + 1, n + 1};  // 1-based indexing
            }

            if (sum < target) {
                p++;
            } else {
                n--;
            }
        }

        return new int[]{-1, -1};
    }
}